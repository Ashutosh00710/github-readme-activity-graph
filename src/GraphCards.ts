import { createGraph } from './createChart';
import { graphSvg } from './svgs';
import { colors, graphArgs, pcsType } from '../interfaces/interface';

export class Card {
  height: number;
  width: number;
  colors: colors;
  title: string;
  area: boolean;
  pcs: pcsType;
  constructor(
    height: number,
    width: number,
    colors: colors,
    title = '',
    area = false,
    pcs: pcsType
  ) {
    this.height = height;
    this.width = width;
    this.colors = colors;
    this.title = title;
    this.area = area;
    this.pcs = pcs;
  }

  async chart(contributions: number[]): Promise<string> {
    //Options to pass in createGraph function
    const options = {
      width: this.width,
      height: this.height,
      axisY: {
        title: 'Contributions',
        onlyInteger: true,
        offset: 70,
        labelOffset: {
          y: 4.5,
        },
      },
      axisX: {
        title: 'Days',
        offset: 50,
        labelOffset: {
          x: -4.5,
        },
      },
      chartPadding: {
        top: 80,
        right: 50,
        bottom: 20,
        left: 20,
      },
      showArea: this.area,
      fullWidth: true,
    };

    //Construction of graph from node-chartist
    const line: Promise<string> = await createGraph('line', options, {
      labels: [...Array(contributions.length + 1).keys()].slice(1),
      series: [{ value: contributions }],
    });

    //Arguments to construct graphs with rect and other options
    const args: graphArgs = {
      height: this.height,
      width: this.width,
      colors: this.colors,
      title: this.title,
      line,
      pcs: this.pcs,
    };

    return graphSvg(args);
  }
}
