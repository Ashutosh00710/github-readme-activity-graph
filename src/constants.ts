export enum Browser {
    SAFARI = 'Safari',
}
