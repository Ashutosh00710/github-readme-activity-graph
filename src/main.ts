import express, { Application } from 'express';
import useragent from 'express-useragent';
import cors from 'cors';
import { Handlers } from './handlers';

const app: Application = express();
const port = process.env.PORT || 5100;

app.use(express.urlencoded({ extended: false }));
app.use(cors());
app.use(useragent.express());

const handlers = new Handlers();

app.get('/', handlers.getRoot);
//Get Graph
app.get('/graph', handlers.getGraph);

app.get('/data', handlers.getData);

app.listen(port, (): void => {
    console.log(`Server is Running on Port ${port}`);
});
