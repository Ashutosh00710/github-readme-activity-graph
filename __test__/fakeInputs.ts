import { Browser } from '../src/constants';
import { GraphArgs } from '../src/interfaces/interface';

export let fakeQueryString = [
    {
        username: 'githubusername',
        radius: 8,
        height: 400,
        hide_title: false,
        custom_title: undefined,
        title_color: 'ffffff',
    },
    {
        username: 'githubusername',
        radius: -10,
        height: 100,
        bg_color: '44475a',
        hide_title: false,
        custom_title: undefined,
    },
    {
        username: 'githubusername',
        radius: 35,
        height: 800,
        bg_color: '44475a',
        color: '000000',
        hide_title: false,
        custom_title: undefined,
    },
    {
        username: 'githubusername',
        bg_color: '44475a',
        color: '000000',
        line: '9e4c98',
        hide_title: false,
        custom_title: undefined,
    },
    {
        username: 'githubusername',
        bg_color: '44475a',
        color: '000000',
        line: '9e4c98',
        point: '44475a',
        hide_title: false,
        custom_title: undefined,
    },
    {
        username: 'githubusername',
        bg_color: '44475a',
        color: 'f8f8f2',
        line: 'ff79c6',
        point: 'bd93f9',
        area: true,
        hide_title: false,
        custom_title: undefined,
    },
    {
        username: 'githubusername',
        bg_color: '44475a',
        color: 'f8f8f2',
        line: 'ff79c6',
        point: 'bd93f9',
        area: true,
        hide_border: true,
        hide_title: false,
        custom_title: undefined,
    },
    {
        username: 'githubusername',
        bg_color: '44475a',
        color: 'f8f8f2',
        line: 'ff79c6',
        point: 'bd93f9',
        area: true,
        hide_border: true,
        hide_title: true,
        custom_title: undefined,
    },
    {
        username: 'githubusername',
        bg_color: '44475a',
        color: 'f8f8f2',
        line: 'ff79c6',
        point: 'bd93f9',
        area: true,
        hide_border: true,
        custom_title: 'some title',
    },
];

export let fakeQueryStringRes = [
    {
        username: 'githubusername',
        radius: 8,
        height: 400,
        colors: {
            areaColor: '9e4c98',
            bgColor: 'ffcfe9',
            borderColor: 'ffffff',
            color: '9e4c98',
            titleColor: 'ffffff',
            lineColor: '9e4c98',
            pointColor: '403d3d',
        },
        hide_title: false,
        area: false,
        days: 31,
        grid: true,
        from: '',
        to: '',
    },
    {
        username: 'githubusername',
        radius: 0,
        height: 200,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: 'ffffff',
            color: '9e4c98',
            titleColor: '9e4c98',
            lineColor: '9e4c98',
            pointColor: '403d3d',
        },
        hide_title: false,
        area: false,
        days: 31,
        from: '',
        grid: true,
        to: '',
    },
    {
        username: 'githubusername',
        radius: 16,
        height: 600,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: 'ffffff',
            color: '000000',
            titleColor: '000000',
            lineColor: '9e4c98',
            pointColor: '403d3d',
        },
        hide_title: false,
        area: false,
        days: 31,
        from: '',
        grid: true,
        to: '',
    },
    {
        username: 'githubusername',
        radius: 0,
        height: 420,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: 'ffffff',
            color: '000000',
            titleColor: '000000',
            lineColor: '9e4c98',
            pointColor: '403d3d',
        },
        hide_title: false,
        area: false,
        days: 31,
        grid: true,
        from: '',
        to: '',
    },
    {
        username: 'githubusername',
        radius: 0,
        height: 420,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: 'ffffff',
            color: '000000',
            titleColor: '000000',
            lineColor: '9e4c98',
            pointColor: '44475a',
        },
        hide_title: false,
        area: false,
        days: 31,
        grid: true,
        from: '',
        to: '',
    },
    {
        username: 'githubusername',
        radius: 0,
        height: 420,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: 'ffffff',
            color: 'f8f8f2',
            titleColor: 'f8f8f2',
            lineColor: 'ff79c6',
            pointColor: 'bd93f9',
        },
        hide_title: false,
        area: true,
        days: 31,
        grid: true,
        from: '',
        to: '',
    },
    {
        username: 'githubusername',
        radius: 0,
        height: 420,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: '0000',
            color: 'f8f8f2',
            titleColor: 'f8f8f2',
            lineColor: 'ff79c6',
            pointColor: 'bd93f9',
        },
        area: true,
        hide_title: false,
        days: 31,
        grid: true,
        from: '',

        to: '',
    },
    {
        username: 'githubusername',
        radius: 0,
        height: 420,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: '0000',
            color: 'f8f8f2',
            titleColor: 'f8f8f2',
            lineColor: 'ff79c6',
            pointColor: 'bd93f9',
        },
        area: true,
        hide_title: true,
        days: 31,
        grid: true,
        from: '',
        to: '',
    },
    {
        username: 'githubusername',
        radius: 0,
        height: 420,
        colors: {
            areaColor: '9e4c98',
            bgColor: '44475a',
            borderColor: '0000',
            color: 'f8f8f2',
            titleColor: 'f8f8f2',
            lineColor: 'ff79c6',
            pointColor: 'bd93f9',
        },
        area: true,
        hide_title: false,
        custom_title: 'some title',
        days: 31,
        grid: true,
        from: '',
        to: '',
    },
];

export let fakeGraphArgsForSafari: GraphArgs = {
    height: 10,
    width: 20,
    radius: 0,
    colors: {
        areaColor: '87ceeb',
        bgColor: '44475a',
        borderColor: '0000',
        color: 'f8f8f2',
        titleColor: 'f8f8f2',
        lineColor: 'ff79c6',
        pointColor: 'bd93f9',
    },
    browser: Browser.SAFARI,
    title: 'graphSvg',
    line: Promise.resolve('line'),
};

export let fakeGraphArgsForBrowsersOtherThanSafari: GraphArgs = {
    height: 10,
    width: 20,
    radius: 0,
    colors: {
        areaColor: '87ceeb',
        bgColor: '44475a',
        borderColor: '0000',
        color: 'f8f8f2',
        titleColor: 'f8f8f2',
        lineColor: 'ff79c6',
        pointColor: 'bd93f9',
    },
    title: 'graphSvg',
    browser: 'Chrome',
    line: Promise.resolve('line'),
};

export let options = {
    width: 10,
    height: 5,
    radius: 5,
    colors: {
        areaColor: 'ff79c6',
        bgColor: '44475a',
        borderColor: 'ffffff',
        color: 'f8f8f2',
        titleColor: 'f8f8f2',
        lineColor: 'ff79c6',
        pointColor: 'bd93f9',
    },
    title: '',
    area: false,
};

export const expectedQuery = (username: string, from: string, to: string) => {
    return {
        query: `
              query userInfo($LOGIN: String!, $FROM: DateTime!, $TO: DateTime!) {
                user(login: $LOGIN) {
                  name
                  contributionsCollection(from: $FROM, to: $TO) {
                    contributionCalendar {
                      weeks {
                        contributionDays {
                          contributionCount
                          date
                        }
                      }
                    }
                  }
                }
              }
            `,
        variables: {
            LOGIN: username,
            FROM: from,
            TO: to,
        },
    };
};

export const dummyWeeksData = [
    {
        contributionDays: [
            {
                contributionCount: 2,
            },
            {
                contributionCount: 3,
            },
            {
                contributionCount: 16,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 5,
            },
            {
                contributionCount: 3,
            },
            {
                contributionCount: 0,
            },
        ],
    },
    {
        contributionDays: [
            {
                contributionCount: 0,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 2,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 5,
            },
        ],
    },
    {
        contributionDays: [
            {
                contributionCount: 1,
            },
            {
                contributionCount: 8,
            },
            {
                contributionCount: 5,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 0,
            },
        ],
    },
    {
        contributionDays: [
            {
                contributionCount: 0,
            },
            {
                contributionCount: 4,
            },
            {
                contributionCount: 8,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 0,
            },
            {
                contributionCount: 14,
            },
            {
                contributionCount: 6,
            },
        ],
    },
    {
        contributionDays: [
            {
                contributionCount: 13,
            },
            {
                contributionCount: 2,
            },
            {
                contributionCount: 10,
            },
            {
                contributionCount: 2,
            },
            {
                contributionCount: 1,
            },
            {
                contributionCount: 5,
            },
            {
                contributionCount: 13,
            },
        ],
    },
    {
        contributionDays: [
            {
                contributionCount: 1,
            },
            {
                contributionCount: 3,
            },
            {
                contributionCount: 6,
            },
            {
                contributionCount: 7,
            },
            {
                contributionCount: 2,
            },
            {
                contributionCount: 1,
            },
            {
                contributionCount: 5,
            },
        ],
    },
    {
        contributionDays: [
            {
                contributionCount: 11,
            },
        ],
    },
];

export const themes = {
    dracula: {
        areaColor: 'ff79c6',
        bgColor: '44475a',
        borderColor: 'ffffff',
        color: 'f8f8f2',
        titleColor: 'f8f8f2',
        lineColor: 'ff79c6',
        pointColor: 'bd93f9',
    },
    gruvbox: {
        areaColor: 'd8a657',
        bgColor: '504945',
        borderColor: 'ffffff',
        color: 'd4be98',
        titleColor: 'd4be98',
        lineColor: 'd8a657',
        pointColor: 'e78a4e',
    },
    github: {
        areaColor: '9ecbff',
        bgColor: '293036',
        borderColor: 'ffffff',
        color: 'ffffff',
        titleColor: 'ffffff',
        lineColor: '9ecbff',
        pointColor: 'f97583',
    },
    rogue: {
        areaColor: 'b18bb1',
        bgColor: '172030',
        borderColor: 'ffffff',
        color: 'a3b09a',
        titleColor: 'a3b09a',
        lineColor: 'b18bb1',
        pointColor: 'c6797e',
    },
    xcode: {
        areaColor: 'c4e3ff',
        bgColor: '202124',
        borderColor: 'ffffff',
        color: 'fcfcfa',
        titleColor: 'fcfcfa',
        lineColor: 'c4e3ff',
        pointColor: 'ff8070',
    },
    redical: {
        areaColor: 'fe428e',
        bgColor: '141321',
        borderColor: 'ffffff',
        color: 'a9fef7',
        titleColor: 'a9fef7',
        lineColor: 'fe428e',
        pointColor: 'f8d847',
    },
    coral: {
        areaColor: 'f4e23d',
        bgColor: '9a3838',
        borderColor: 'ffffff',
        color: 'f9fae9',
        titleColor: 'f9fae9',
        lineColor: 'f4e23d',
        pointColor: 'f4e7e7',
    },
    reactDark: {
        areaColor: '5bcdec',
        bgColor: '0d1117',
        borderColor: 'ffffff',
        color: '5bcdec',
        titleColor: '5bcdec',
        lineColor: '5bcdec',
        pointColor: 'ffffff',
    },
    nord: {
        areaColor: '88c0d0',
        bgColor: '2e3440',
        borderColor: 'ffffff',
        color: '88c0d0',
        titleColor: '88c0d0',
        lineColor: '88c0d0',
        pointColor: 'ffffff',
    },
    lucent: {
        areaColor: 'ffd369',
        bgColor: 'cccccc',
        borderColor: '000000',
        color: '000000',
        titleColor: '000000',
        lineColor: 'ffd369',
        pointColor: 'faf3e0',
    },
    chartreuseDark: {
        areaColor: '00adfe',
        bgColor: '000000',
        borderColor: '000000',
        color: '7ffe00',
        titleColor: '7ffe00',
        lineColor: '00adfe',
        pointColor: '7ffe00',
    },
    githubLight: {
        areaColor: '9be9a8',
        bgColor: 'ffffff',
        borderColor: 'ffffff',
        color: '000000',
        titleColor: '000000',
        lineColor: '9be9a8',
        pointColor: '40c463',
    },
    minimal: {
        areaColor: 'd3e6fa',
        bgColor: 'ffffff',
        borderColor: 'ffffff',
        color: '000000',
        titleColor: '000000',
        lineColor: 'd3e6fa',
        pointColor: '3f99ed',
    },
    materialPalenight: {
        areaColor: 'c792ea',
        bgColor: '292d3e',
        borderColor: 'ffffff',
        color: 'a6accd',
        titleColor: 'a6accd',
        lineColor: 'c792ea',
        pointColor: '89ddff',
    },
    green: {
        areaColor: '588157',
        borderColor: 'ffffff',
        bgColor: 'dad7cd',
        color: '3a5a40',
        titleColor: '3a5a40',
        lineColor: '588157',
        pointColor: '344e41',
    },
    gotham: {
        areaColor: '2aa889',
        bgColor: '0c1014',
        borderColor: '2aa889',
        color: '2aa889',
        titleColor: '2aa889',
        lineColor: '599cab',
        pointColor: '99d1ce',
    },
    noctis_minimus: {
        areaColor: '72b7c0',
        borderColor: 'ffffff',
        bgColor: '1b2932',
        color: 'd3b692',
        titleColor: 'd3b692',
        lineColor: '72b7c0',
        pointColor: 'c5cdd3',
    },
    one_dark: {
        areaColor: 'e5c17c',
        borderColor: 'ffffff',
        bgColor: '282C34',
        color: 'abb2bf',
        titleColor: 'abb2bf',
        lineColor: 'e5c17c',
        pointColor: 'e06c75',
    },
    monokai: {
        areaColor: 'ff6188',
        borderColor: 'ffffff',
        bgColor: '2D2A2E',
        color: 'fcfcfa',
        titleColor: 'fcfcfa',
        lineColor: 'ff6188',
        pointColor: 'ffd866',
    },
    'synthwave-84': {
        areaColor: '3A2442',
        borderColor: '251e30',
        bgColor: '2C223B',
        color: 'FF3CA2',
        titleColor: 'FF3CA2',
        lineColor: 'F7F645',
        pointColor: '22E5F4',
    },
    aqua: {
        areaColor: '226d64',
        borderColor: '161e2d',
        bgColor: '52b69a',
        color: '187177',
        titleColor: '187177',
        lineColor: '226d64',
        pointColor: '156f69',
    },
    default: {
        areaColor: '9e4c98',
        bgColor: 'ffcfe9',
        borderColor: 'ffffff',
        color: '9e4c98',
        titleColor: '9e4c98',
        lineColor: '9e4c98',
        pointColor: '403d3d',
    },
};
