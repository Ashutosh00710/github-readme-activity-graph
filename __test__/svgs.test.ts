import { graphSvg, invalidUserSvg } from '../src/svgs';
import { Card } from '../src/GraphCards';
import { fakeGraphArgs, fakeQueryStringRes } from './fakeInputs';

//- Svg testing ✔
test('Test SVGs', () => {
  let fakeInvalidSvgArg = 'User not found!';
  expect.assertions(4);
  expect(graphSvg(fakeGraphArgs)).toEqual(expect.any(String));
  expect(graphSvg(fakeGraphArgs)).toMatchSnapshot();
  expect(invalidUserSvg(fakeInvalidSvgArg)).toEqual(expect.any(String));
  expect(invalidUserSvg(fakeInvalidSvgArg)).toMatchSnapshot();
});

//- Svg testing Promise<string> (GrapgCards.ts)✔
test('chart SVGs', async () => {
  expect.assertions(1);
  expect(
    await new Card(
      420,
      1200,
      fakeQueryStringRes[0].colors,
      "xyz's Contribution Graph",
      fakeQueryStringRes[0].area,
      fakeQueryStringRes[0].pcs
    ).chart([1, 2, 3, 4, 5])
  ).toMatchSnapshot();
});
