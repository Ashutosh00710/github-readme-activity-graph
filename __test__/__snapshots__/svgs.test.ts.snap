// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SVGs 1`] = `
"
    <svg
        width=\\"20\\"
        height=\\"10\\"
        viewBox=\\"0 0 20 10\\"
        fill=\\"none\\"
        xmlns=\\"http://www.w3.org/2000/svg\\">
            <rect xmlns=\\"http://www.w3.org/2000/svg\\" data-testid=\\"card_bg\\" id=\\"cardBg\\"
            x=\\"0\\" y=\\"0\\" rx=\\"2.5\\" height=\\"100%\\" stroke=\\"#E4E2E2\\" fill-opacity=\\"1\\"
            width=\\"100%\\" fill=\\"#44475a\\" stroke-opacity=\\"1\\" style=\\"stroke:#0000; stroke-width:1;\\"/>

            <style>
                body {
                    font: 600 18px 'Segoe UI', Ubuntu, Sans-Serif;
                }
                .header {
                    font: 600 18px 'Segoe UI', Ubuntu, Sans-Serif;
                    text-align: center;
                    color: #f8f8f2;
                    margin-top: 20px;
                }
                svg {
                    font: 600 18px 'Segoe UI', Ubuntu, Sans-Serif;
                }
                
    .ct-label {
      fill: #f8f8f2;
      color: #f8f8f2;
      font-size: .75rem;
      line-height: 1;
    }

    .ct-grid-background,
    .ct-line {
      fill: none;
    }

    .ct-chart-bar .ct-label,
    .ct-chart-line .ct-label {
      display: block;
      display: -webkit-box;
      display: -moz-box;
      display: -ms-flexbox;
      display: -webkit-flex;
      display: flex;
    }

    .ct-label.ct-horizontal.ct-start {
      -webkit-box-align: flex-end;
      -webkit-align-items: flex-end;
      -ms-flex-align: flex-end;
      align-items: flex-end;
      -webkit-box-pack: flex-start;
      -webkit-justify-content: flex-start;
      -ms-flex-pack: flex-start;
      justify-content: flex-start;
      text-align: left;
      text-anchor: start;
    }

    .ct-label.ct-horizontal.ct-end {
      -webkit-box-align: flex-start;
      -webkit-align-items: flex-start;
      -ms-flex-align: flex-start;
      align-items: flex-start;
      -webkit-box-pack: flex-start;
      -webkit-justify-content: flex-start;
      -ms-flex-pack: flex-start;
      justify-content: flex-start;
      text-align: left;
      text-anchor: start;
    }

    .ct-label.ct-vertical.ct-start {
      -webkit-box-align: flex-end;
      -webkit-align-items: flex-end;
      -ms-flex-align: flex-end;
      align-items: flex-end;
      -webkit-box-pack: flex-end;
      -webkit-justify-content: flex-end;
      -ms-flex-pack: flex-end;
      justify-content: flex-end;
      text-align: right;
      text-anchor: end;
    }

    .ct-label.ct-vertical.ct-end {
      -webkit-box-align: flex-end;
      -webkit-align-items: flex-end;
      -ms-flex-align: flex-end;
      align-items: flex-end;
      -webkit-box-pack: flex-start;
      -webkit-justify-content: flex-start;
      -ms-flex-pack: flex-start;
      justify-content: flex-start;
      text-align: left;
      text-anchor: start;
    }

    .ct-grid {
      stroke: #f8f8f2;
      stroke-width: 1px;
      stroke-opacity: 0.3;
      stroke-dasharray: 2px;
    }

    .ct-point {
      stroke-width: 10px;
      stroke-linecap: round;
      stroke: #bd93f9;
      animation: blink 1s ease-in-out forwards;
    }

    .ct-line {
      stroke-width: 4px;
      stroke-dasharray: 5000;
      stroke-dashoffset: 5000;
      stroke: #ff79c6;
      animation: dash 5s ease-in-out forwards;
    }

    .ct-area {
      stroke: none;
      fill-opacity: 0.1;
    }

    .ct-series-a .ct-area,
    .ct-series-a .ct-slice-pie {
      fill: #87ceeb;
    }

    .ct-label .ct-horizontal {
      transform: rotate(-90deg)
    }

    .ct-label.ct-horizontal.ct-end {
      transform: rotate(-30deg);
      -webkit-transform: rotate(-30deg);
      -moz-transform: rotate(-30deg);
      -o-transform: rotate(-30deg);
      -ms-transform: rotate(-30deg);
      transform-box: fill-box;
      transform-origin: 100% 0;
      text-anchor: end;
    }

    
                
        @keyframes blink {
            from {
                opacity: 0;
                transform:translateX(-20px); 
            }
            to {
                opacity:1;
                transform: translateX(0);
            }
        }
    
                
        @keyframes dash {
            to {
                stroke-dashoffset: 0;
            }
        }
    
    
            </style>

            <foreignObject x=\\"0\\" y=\\"0\\" width=\\"20\\" height=\\"50\\">
                <h1 xmlns=\\"http://www.w3.org/1999/xhtml\\" class=\\"header\\">
                    graphSvg
                </h1>
            </foreignObject>
            [object Promise]
    </svg>
"
`;

exports[`Test SVGs 2`] = `
"
    <svg
        width=\\"400\\"
        height=\\"200\\"
        viewBox=\\"0 0 420 200\\"
        fill=\\"none\\"
        xmlns=\\"http://www.w3.org/2000/svg\\">
        <style>
                svg {
                    font: 600 14px 'Segoe UI', Ubuntu, Sans-Serif;
                }
        </style>
        <rect xmlns=\\"http://www.w3.org/2000/svg\\" data-testid=\\"card_bg\\" id=\\"cardBg\\" x=\\"0.5\\"
        y=\\"0.5\\" rx=\\"4.5\\" height=\\"100%\\" stroke=\\"#E4E2E2\\" fill-opacity=\\"1\\" width=\\"100%\\"
        fill=\\"#44475a\\" stroke-opacity=\\"1\\"/>
        <text x=\\"20\\" y=\\"100\\" fill=\\"#bd93f9\\">User not found!</text>
    </svg>
"
`;

exports[`chart SVGs 1`] = `
"
    <svg
        width=\\"1200\\"
        height=\\"420\\"
        viewBox=\\"0 0 1200 420\\"
        fill=\\"none\\"
        xmlns=\\"http://www.w3.org/2000/svg\\">
            <rect xmlns=\\"http://www.w3.org/2000/svg\\" data-testid=\\"card_bg\\" id=\\"cardBg\\"
            x=\\"0\\" y=\\"0\\" rx=\\"2.5\\" height=\\"100%\\" stroke=\\"#E4E2E2\\" fill-opacity=\\"1\\"
            width=\\"100%\\" fill=\\"#ffcfe9\\" stroke-opacity=\\"1\\" style=\\"stroke:#ffffff; stroke-width:1;\\"/>

            <style>
                body {
                    font: 600 18px 'Segoe UI', Ubuntu, Sans-Serif;
                }
                .header {
                    font: 600 18px 'Segoe UI', Ubuntu, Sans-Serif;
                    text-align: center;
                    color: #9e4c98;
                    margin-top: 20px;
                }
                svg {
                    font: 600 18px 'Segoe UI', Ubuntu, Sans-Serif;
                }
                
    .ct-label {
      fill: #9e4c98;
      color: #9e4c98;
      font-size: .75rem;
      line-height: 1;
    }

    .ct-grid-background,
    .ct-line {
      fill: none;
    }

    .ct-chart-bar .ct-label,
    .ct-chart-line .ct-label {
      display: block;
      display: -webkit-box;
      display: -moz-box;
      display: -ms-flexbox;
      display: -webkit-flex;
      display: flex;
    }

    .ct-label.ct-horizontal.ct-start {
      -webkit-box-align: flex-end;
      -webkit-align-items: flex-end;
      -ms-flex-align: flex-end;
      align-items: flex-end;
      -webkit-box-pack: flex-start;
      -webkit-justify-content: flex-start;
      -ms-flex-pack: flex-start;
      justify-content: flex-start;
      text-align: left;
      text-anchor: start;
    }

    .ct-label.ct-horizontal.ct-end {
      -webkit-box-align: flex-start;
      -webkit-align-items: flex-start;
      -ms-flex-align: flex-start;
      align-items: flex-start;
      -webkit-box-pack: flex-start;
      -webkit-justify-content: flex-start;
      -ms-flex-pack: flex-start;
      justify-content: flex-start;
      text-align: left;
      text-anchor: start;
    }

    .ct-label.ct-vertical.ct-start {
      -webkit-box-align: flex-end;
      -webkit-align-items: flex-end;
      -ms-flex-align: flex-end;
      align-items: flex-end;
      -webkit-box-pack: flex-end;
      -webkit-justify-content: flex-end;
      -ms-flex-pack: flex-end;
      justify-content: flex-end;
      text-align: right;
      text-anchor: end;
    }

    .ct-label.ct-vertical.ct-end {
      -webkit-box-align: flex-end;
      -webkit-align-items: flex-end;
      -ms-flex-align: flex-end;
      align-items: flex-end;
      -webkit-box-pack: flex-start;
      -webkit-justify-content: flex-start;
      -ms-flex-pack: flex-start;
      justify-content: flex-start;
      text-align: left;
      text-anchor: start;
    }

    .ct-grid {
      stroke: #9e4c98;
      stroke-width: 1px;
      stroke-opacity: 0.3;
      stroke-dasharray: 2px;
    }

    .ct-point {
      stroke-width: 10px;
      stroke-linecap: round;
      stroke: #403d3d;
      animation: blink 1s ease-in-out forwards;
    }

    .ct-line {
      stroke-width: 4px;
      stroke-dasharray: 5000;
      stroke-dashoffset: 5000;
      stroke: #9e4c98;
      animation: dash 5s ease-in-out forwards;
    }

    .ct-area {
      stroke: none;
      fill-opacity: 0.1;
    }

    .ct-series-a .ct-area,
    .ct-series-a .ct-slice-pie {
      fill: #9e4c98;
    }

    .ct-label .ct-horizontal {
      transform: rotate(-90deg)
    }

    .ct-label.ct-horizontal.ct-end {
      transform: rotate(-30deg);
      -webkit-transform: rotate(-30deg);
      -moz-transform: rotate(-30deg);
      -o-transform: rotate(-30deg);
      -ms-transform: rotate(-30deg);
      transform-box: fill-box;
      transform-origin: 100% 0;
      text-anchor: end;
    }

    
                
        @keyframes blink {
            from {
                opacity: 0;
                transform:translateX(-20px); 
            }
            to {
                opacity:1;
                transform: translateX(0);
            }
        }
    
                
        @keyframes dash {
            to {
                stroke-dashoffset: 0;
            }
        }
    
    
            </style>

            <foreignObject x=\\"0\\" y=\\"0\\" width=\\"1200\\" height=\\"50\\">
                <h1 xmlns=\\"http://www.w3.org/1999/xhtml\\" class=\\"header\\">
                    xyz's Contribution Graph
                </h1>
            </foreignObject>
            <svg xmlns:ct=\\"http://gionkunz.github.com/chartist-js/ct\\" width=\\"1200\\" height=\\"420\\" class=\\"ct-chart-line\\"><g class=\\"ct-grids\\"><line x1=\\"90\\" x2=\\"90\\" y1=\\"80\\" y2=\\"330\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"355\\" x2=\\"355\\" y1=\\"80\\" y2=\\"330\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"620\\" x2=\\"620\\" y1=\\"80\\" y2=\\"330\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"885\\" x2=\\"885\\" y1=\\"80\\" y2=\\"330\\" class=\\"ct-grid ct-horizontal\\"></line><line x1=\\"1150\\" x2=\\"1150\\" y1=\\"80\\" y2=\\"330\\" class=\\"ct-grid ct-horizontal\\"></line><line y1=\\"330\\" y2=\\"330\\" x1=\\"90\\" x2=\\"1150\\" class=\\"ct-grid ct-vertical\\"></line><line y1=\\"267.5\\" y2=\\"267.5\\" x1=\\"90\\" x2=\\"1150\\" class=\\"ct-grid ct-vertical\\"></line><line y1=\\"205\\" y2=\\"205\\" x1=\\"90\\" x2=\\"1150\\" class=\\"ct-grid ct-vertical\\"></line><line y1=\\"142.5\\" y2=\\"142.5\\" x1=\\"90\\" x2=\\"1150\\" class=\\"ct-grid ct-vertical\\"></line><line y1=\\"80\\" y2=\\"80\\" x1=\\"90\\" x2=\\"1150\\" class=\\"ct-grid ct-vertical\\"></line></g><g><g class=\\"ct-series ct-series-a\\"><path d=\\"M90,330C178.333,309.167,266.667,288.333,355,267.5C443.333,246.667,531.667,225.833,620,205C708.333,184.167,796.667,163.333,885,142.5C973.333,121.667,1061.667,100.833,1150,80\\" class=\\"ct-line\\"></path><line x1=\\"90\\" y1=\\"330\\" x2=\\"90.01\\" y2=\\"330\\" class=\\"ct-point\\" ct:value=\\"1\\"></line><line x1=\\"355\\" y1=\\"267.5\\" x2=\\"355.01\\" y2=\\"267.5\\" class=\\"ct-point\\" ct:value=\\"2\\"></line><line x1=\\"620\\" y1=\\"205\\" x2=\\"620.01\\" y2=\\"205\\" class=\\"ct-point\\" ct:value=\\"3\\"></line><line x1=\\"885\\" y1=\\"142.5\\" x2=\\"885.01\\" y2=\\"142.5\\" class=\\"ct-point\\" ct:value=\\"4\\"></line><line x1=\\"1150\\" y1=\\"80\\" x2=\\"1150.01\\" y2=\\"80\\" class=\\"ct-point\\" ct:value=\\"5\\"></line></g></g><g class=\\"ct-labels\\"><text x=\\"85.5\\" y=\\"350\\" width=\\"265\\" height=\\"60\\" class=\\"ct-label ct-horizontal ct-end\\"></text><text x=\\"350.5\\" y=\\"350\\" width=\\"265\\" height=\\"60\\" class=\\"ct-label ct-horizontal ct-end\\"></text><text x=\\"615.5\\" y=\\"350\\" width=\\"265\\" height=\\"60\\" class=\\"ct-label ct-horizontal ct-end\\"></text><text x=\\"880.5\\" y=\\"350\\" width=\\"265\\" height=\\"60\\" class=\\"ct-label ct-horizontal ct-end\\"></text><text x=\\"1145.5\\" y=\\"350\\" width=\\"30\\" height=\\"60\\" class=\\"ct-label ct-horizontal ct-end\\"></text><text y=\\"334.5\\" x=\\"80\\" height=\\"62.5\\" width=\\"60\\" class=\\"ct-label ct-vertical ct-start\\">1</text><text y=\\"272\\" x=\\"80\\" height=\\"62.5\\" width=\\"60\\" class=\\"ct-label ct-vertical ct-start\\">2</text><text y=\\"209.5\\" x=\\"80\\" height=\\"62.5\\" width=\\"60\\" class=\\"ct-label ct-vertical ct-start\\">3</text><text y=\\"147\\" x=\\"80\\" height=\\"62.5\\" width=\\"60\\" class=\\"ct-label ct-vertical ct-start\\">4</text><text y=\\"84.5\\" x=\\"80\\" height=\\"30\\" width=\\"60\\" class=\\"ct-label ct-vertical ct-start\\">5</text></g><text class=\\"ct-axis-title ct-label\\" x=\\"620\\" y=\\"400\\" dominant-baseline=\\"text-after-edge\\" text-anchor=\\"middle\\">Dates</text><text class=\\"ct-axis-title ct-label\\" x=\\"20\\" y=\\"205\\" transform=\\"rotate(-90, 20, 205)\\" dominant-baseline=\\"hanging\\" text-anchor=\\"middle\\">Contributions</text></svg>
    </svg>
"
`;
